lib/CapabilityDelegation.js

```diff
@@ -40,11 +40,15 @@ module.exports = class CapabilityDelegation extends ControllerProofPurpose {
    *   the date on the signature can deviate from, defaults to `Infinity`.
    * @param {function} [inspectCapabilityChain] - See documentation for
    *   `utils.verifyCapabilityChain`.
+   * @param [currentDate = new Date()] {Date} - The date used for comparison
+   *   when determining if a capability has expired.
+   * @param [maxChainLength=10] - The maximum length of the capability
+   *   delegation chain.
    */
   constructor({
     capabilityChain, verifiedParentCapability,
     expectedTarget, expectedRootCapability, inspectCapabilityChain,
-    capability, capabilityAction, caveat, suite,
+    capability, capabilityAction, caveat, suite, currentDate, maxChainLength,
```

Comment:
The docs say default `maxChainLength=10` ... so did we want this? Same thing appears in a few other places.

```suggestion
    capability, capabilityAction, caveat, suite, currentDate, maxChainLength = 10,
```

Hmm, I see it's set to `constants.MAX_CHAIN_LENGTH` elsewhere ... so maybe this is ok at this layer? Also, do we support having the docs say something like `constants.MAX_CHAIN_LENGTH` and then we define that elsewhere -- or should that be used internally only (slight worry about things getting out of sync, maybe not important).

Reply:
The concern about sync is what informed the current code.  I don't know what the best solutions is.  Putting `constants.MAX_CHAIN_LENGTH`in the jsdocs is very unhelpful if one would like to know what the value is.  And putting a hard coded value has the potential to get out of sync with the code.